docs: refactor/flesh out docker docs

ajcwebdev opened the pull request:
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-08-15).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

- Added more headers for easier side bar navigation
- Fixed `build` command which was using `-e` incorrectly.
- Added instructions for deploying to Railway
- Added [repo with working Docker setup](https://github.com/ajcwebdev/ct3a-docker)
- Added [a link to a running example on Railway](https://ct3a-docker-production.up.railway.app/)
- Added more resources for specifically Next.js+Docker

---

## Screenshots

<img width="1254" alt="screenshot of docker docs" src="https://user-images.githubusercontent.com/12433465/191856681-d585e442-d2ed-4115-b90d-a608b9b86b6f.png">

💯

juliusmarminge commented on www/src/pages/en/deployment/docker.md:
```diff
@@ -4,187 +4,195 @@ description: Deployment with Docker
 ---
 
-You can containerize this stack and deploy it as a single container using Docker, or as a part of a group of containers using docker-compose.
+You can containerize this stack and deploy it as a single container using Docker, or as a part of a group of containers using docker-compose. See [`ct3a-docker`](https://github.com/ajcwebdev/ct3a-docker) for an example repo based on this doc.
```
You can self-plug here if you'd like ;)

```suggestion
You can containerize this stack and deploy it as a single container using Docker, or as a part of a group of containers using docker-compose. See [`ajcwebdev/ct3a-docker`](https://github.com/ajcwebdev/ct3a-docker) for an example repo based on this doc.
```

juliusmarminge commented on www/src/pages/en/deployment/docker.md:
```diff
@@ -4,187 +4,195 @@ description: Deployment with Docker
+### 2. Remove Env Import
 
-    <details>
+Remove the `env`-import from [`next.config.mjs`](https://github.com/t3-oss/create-t3-app/blob/main/cli/template/base/next.config.mjs):
```
we've been asked why you need to do this so maybe we should add something here regarding this being pulled in to the build image or smth 🤔

juliusmarminge commented on cli/template/base/README.md:
```diff
@@ -41,185 +41,7 @@ We recommend deploying to [Vercel](https://vercel.com/?utm_source=t3-oss&utm_cam
-- [Compose file version 3 reference](https://docs.docker.com/compose/compose-file/compose-file-v3/)
-- [Docker CLI reference](https://docs.docker.com/engine/reference/commandline/docker/)
-- [Docker Compose CLI reference](https://docs.docker.com/compose/reference/)
+You can also dockerize this stack and deploy a container. See the [Docker deployment page](https://github.com/t3-oss/create-t3-app/blob/main/www/src/pages/en/deployment/docker.md) for details.
```
link to the actual docs?

```suggestion
You can also dockerize this stack and deploy a container. See the [Docker deployment page](https://create-t3-app-nu.vercel.app/en/deployment/docker) for details.
```

domain will change though so we'd have to remember to switch it

ajcwebdev replied:
Yeah that was my thought with including the GitHub link, but if you think you'll remember whenever the switch happens then it won't be an issue.

juliusmarminge replied:
hopefully wont be too much longer haha but yea

nexxeln reviewed (commented):
I liked the details tag for the Dockerfile. It's too long.

ajcwebdev commented:
Alright, I don't think it's inline with the intent of the HTML tags themselves, and it's not something you'll see often on documentation sites. Usually these are used for extra notes that could be skipped and would be unimportant to the greater flow of the content explained on that page. But I'll defer to you for final say cause it's your repo, so I've added them back in (and this time there aren't extra HTML tags leaking through).

@juliusmarminge I'm going to have to do some digging in the Astro docs and probably ask around to find out how they're implementing the titles on the code blocks. I think it's an `mdx` thing that needs to be set correctly in the `astro.config.mjs` file. For now I've removed the comments with the file names and I can open another PR for the whole site once I figure out how to get the file name headers configured.
